neutronclient/client.py

```diff
@@ -239,6 +248,8 @@ def authenticate(self):
             self._authenticate_keystone()
         elif self.auth_strategy == 'noauth':
             self._authenticate_noauth()
+        elif self.auth_plugin:
+            self._plugin_auth(self.auth_url)
```

Comment:
If auth-strategy is unset it'll be keystone. as a user I feel like if I installed a plugin it'd just be natural to be what happens by default.